### ISSUE TITLE
Deployment doesn't respect choice of node v20 based on engines.node and errors out

### ISSUE THREAD

**Issue by mcmxcdev** [zeit/now#10850]
According to https://vercel.com/docs/functions/serverless-functions/runtimes/node-js#node.js-version, it should be possible to define engines.node as either `20.x`, `>20.0.0`, or `20.5.1` which then overrules whatever version was picked inside the Vercel UI dropdown for Node.js

This doesn't seem to work as expected, I have tried these variations leading to build failures:
```
Warning: Due to "engines": { "node": "20.5.1" } in your `package.json` file, the Node.js Version defined in your Project Settings ("18.x") will not apply. Learn More: http://vercel.link/node-version
Error: Found invalid Node.js Version: "20.5.1". Please set "engines": { "node": "18.x" } in your `package.json` file to use Node.js 18.
```

```
Warning: Due to "engines": { "node": "20.x" } in your `package.json` file, the Node.js Version defined in your Project Settings ("18.x") will not apply. Learn More: http://vercel.link/node-version
Error: Found invalid Node.js Version: "20.x". Please set "engines": { "node": "18.x" } in your `package.json` file to use Node.js 18.
```

```
Warning: Due to "engines": { "node": "^20.0.0" } in your `package.json` file, the Node.js Version defined in your Project Settings ("18.x") will not apply. Learn More: http://vercel.link/node-version
Error: Found invalid Node.js Version: "^20.0.0". Please set "engines": { "node": "18.x" } in your `package.json` file to use Node.js 18.
```

Since we have a team of devs working on the same codebase, I can't simply change the used node version in Vercel UI first and then afterwards update `engines.node`, as we have pre-existing engine restrictions for v18 still in place.

**Comment by Tobi-mmt** [comment]
I can confirm the issue. The vercel UI has a different set of allowed node versions than the `engines.node` field.

**Comment by rtrvrtg** [comment]
I'm also seeing this. It doesn't let me run Node 20 builds until I set the default version to Node 20.

**Comment by jaredcbaum** [comment]
https://vercel.com/docs/functions/serverless-functions/runtimes/node-js#version-overrides-in-package.json

The docs say that the `engines.node` field **dictates** which version may be selected from the Vercel UI dropdown. This tells me that you cannot purely override what is selected in the dropdown from the `engines.node` field.

**Comment by wilson-webdev** [comment]
I am also seeing this. I have this in my `package.json`

```json
"engines": {
  "node": "20.x"
}
```

and this is the error I am receiving: `Error: Found invalid Node.js Version: "20.x". Please set "engines": { "node": "18.x" } in your package.json file to use Node.js 18.`

**Comment by Aetrix27** [comment]
I am also having this issue

**Comment by ParallelUniv3rse** [comment]
I've had the dsame issue and contacted Vercel support about it, bringing their attention to this thread. 

TL;DR 
I've set the version in my project settings to node 20.x and deployed. Vercel support assured me that node 16 (and perhaps node 18) deploys should still work fine when lower versions of node are specified in the `engines` in package.json. 

Messages with support: 

```
Thanks for reaching out to Vercel Support!

You've manually specified a fixed Node.js version in your `package.json`, which isn't necessarily available in our build container.
However, as outlined in our documentation, you cannot force to have a newer version than specified in your Project settings (see documentation: https://vercel.com/docs/functions/runtimes/node-js#version-overrides-in-package.json ). 

There are two ways for you how you could test a Node.js 20 setup:
1. You could set Node.js 20.x in your project settings and the `engines` section of your `package.json` to `16.x` for your current deployments (switching the Node.js version doesn't affect existing deployments, only new deployments will be affected).

2. You can create a new project with Node.js 20.x set as the Node.js version – this wouldn't affect your existing production project.
```

```
I have updates to node 20.x in the project settings and got it working that way.

One concern still remains - can I now deploy node 16 apps to production when I have node 20.x in the project settings, but node 16.x in the `engines` in package.json ?
```
```
This should not cause any issues and continue to work as expected.
In order to be sure, it's recommend to create another preview deployment with this setting first, by branching off your `main` branch and deploying without the `--prod` flag. 

Please let us know if you have any further questions or issues.
```

**Comment by ParallelUniv3rse** [comment]
update: Seems like the support was not right. 
Trying to deploy 16.x with 20.x defined in the project settings breaks the build with the following output: 

```

Vercel CLI 33.5.4
--
17:56:11.191 | Warning: Due to "engines": { "node": "^16.10.0" } in your `package.json` file, the Node.js Version defined in your Project Settings ("20.x") will not apply. Learn More: http://vercel.link/node-version
17:56:11.196 | Error: Found invalid Node.js Version: "^16.10.0". Please set "engines": { "node": "20.x" } in your `package.json` file to use Node.js 20.
17:56:11.196 | Learn More: http://vercel.link/node-version

```

**Comment by tommymarshall** [comment]
I wonder if it has to do with the Project > Settings Node version to 20 ahead of time? The one issue is you'd have to change that setting and then be sure not to do a production deploy, do some release branches so your Preview's are deployed using the correct Node version.

**Comment by ofhouse** [comment]
Hey folks, thank you for raising this issue!

It took us some time to figure out a solution for this as with Node 20 we switched to a newer build image for the CI part (See the [build image section](https://vercel.com/docs/deployments/build-image) in our docs for more information).
This way it was not possible to override the node version to our latest `20.x` runtime when the project setting was set to `18.x` or `16.x`.

The good news is that we finally found a solution and with the latest [CLI 33.6.1 release](https://github.com/vercel/vercel/releases/tag/vercel%4033.6.1) this should no longer be a problem.

We also updated our documentation on that: [Version overrides in package.json](https://vercel.com/docs/functions/runtimes/node-js#version-overrides-in-package.json)

Have a nice weekend!
